services/provider/server/server.go

```diff
@@ -5,43 +5,202 @@ import (
 	"fmt"
 	"net"
 
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
 	pb "github.com/red-hat-storage/ocs-operator/services/provider/pb"
+	"github.com/red-hat-storage/ocs-operator/services/provider/server/consumer"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog"
 )
 
-type ocsProviderServer struct {
+type OCSProviderServer struct {
 	pb.UnimplementedOCSProviderServer
+	consumerAPI consumer.APIUpdater
+	authManager *AuthManager
+	namespace   string
+}
+
+func NewOCSProviderServer(cClient consumer.APIUpdater, auth *AuthManager, namespace string) *OCSProviderServer {
+	return &OCSProviderServer{
+		consumerAPI: cClient,
+		authManager: auth,
+		namespace:   namespace,
+	}
+}
+
+// GenerateToken RPC call to generate a new jwt token for the consumer cluster
+func (s *OCSProviderServer) GenerateToken(ctx context.Context, req *pb.GenerateTokenRequest) (*pb.GenerateTokenResponse, error) {
+	token, err := s.authManager.GetAccessToken()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to generate access token: %v", err)
+	}
+	return &pb.GenerateTokenResponse{AccessToken: token}, nil
```

Comment:
Why do we have this? We should not provide a public endpoint for token generation